#!/usr/bin/env python
PACKAGE = "hfts_grasp_planner"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add('min_iterations', int_t, 0, "Minimum number iterations", 20)
gen.add('max_iterations', int_t, 0, "Maximum number iterations", 70)
gen.add('free_space_weight', double_t, 0, "Weight for free space proximity", 0.5, 0.0, 10.0)
gen.add('connected_space_weight', double_t, 0, "Weight for connected space proximity", 4.0, 0.0, 10.0)
gen.add('time_limit', double_t, 0, "Maximal planning time is seconds", 60.0, 0.0, 6000.0)
gen.add('use_approximates', bool_t, 0, "Flag whether to use approximate goals", True)
gen.add('compute_velocities', bool_t, 0, "Flag whether to compute velocities", True)

exit(gen.generate(PACKAGE, "hfts_integrated_planner_node", "IntegratedHFTSPlanner"))